src/main/java/uk/gov/hmcts/reform/bailcaseapi/domain/service/MakeNewApplicationService.java

```diff
@@ -0,0 +1,202 @@
+package uk.gov.hmcts.reform.bailcaseapi.domain.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bailcaseapi.domain.UserDetailsHelper;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.BailCase;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.BailCaseFieldDefinition;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.PriorApplication;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.UserDetails;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.UserRoleLabel;
+import uk.gov.hmcts.reform.bailcaseapi.domain.entities.ccd.field.IdValue;
+
+@Service
+public class MakeNewApplicationService {
+
+    @Autowired
+    private Appender<PriorApplication> appender;
+
+    @Autowired
+    private UserDetails userDetails;
+
+    @Autowired
+    private UserDetailsHelper userDetailsHelper;
+
+    private final ObjectMapper mapper;
+
+    public MakeNewApplicationService(Appender<PriorApplication> appender,
+                                     UserDetails userDetails, UserDetailsHelper userDetailsHelper,
+                                     ObjectMapper mapper) {
+        this.appender = appender;
+        this.userDetails = userDetails;
+        this.userDetailsHelper = userDetailsHelper;
+        this.mapper = mapper;
+    }
+
+    public void appendPriorApplication(BailCase bailCase, BailCase bailCaseBefore) {
+
+        Optional<List<IdValue<PriorApplication>>> maybeExistingPriorApplication =
+            bailCaseBefore.read(BailCaseFieldDefinition.PRIOR_APPLICATIONS);
+
+        String nextAppId = String.valueOf(maybeExistingPriorApplication.orElse(Collections.emptyList()).size() + 1);
+
+        String previousCaseDataJson;
+
+        try {
+            previousCaseDataJson = mapper.writeValueAsString(bailCaseBefore);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("Could not serialize data", e);
+        }
+
+        List<IdValue<PriorApplication>> allPriorApplications = appender.append(
+            buildNewPriorApplication(nextAppId, previousCaseDataJson),
+            maybeExistingPriorApplication.orElse(Collections.emptyList()));
+
+        bailCase.write(BailCaseFieldDefinition.PRIOR_APPLICATIONS, allPriorApplications);
+
+    }
+
+    public void clearUnrelatedFields(BailCase bailCase) {
+        bailCase.entrySet().removeIf(entry -> !validFields.contains(entry.getKey()));
```

Comment:
Can we add a remove method in bailcase, to do this instead of having the logic here?